platforms/qhaikuplatform/haiku/qhaikuscreen.cpp

```diff
@@ -58,6 +59,9 @@ QHaikuScreen::QHaikuScreen()
 	, m_screen(new BScreen(B_MAIN_SCREEN_ID))
 {
 	Q_ASSERT(m_screen->IsValid());
+
+	// Set a global scale factor instead of a per-screen one (for now.)
+	QHighDpiScaling::setGlobalFactor(std::max(1.0f, be_plain_font->Size() / 12.0f));
```

Comment:
This is likely to break various applications as this force overrides the scale factor, it won't respect the [high-DPI scale factor rounding policy](https://doc.qt.io/qt-5/qguiapplication.html#highDpiScaleFactorRoundingPolicy) set by application anymore. You should implement QHaikuScreen::logicalDpi and QHaikuScreen::logicalBaseDpi instead. You can send DPI change event with QWindowSystemInterface::handleScreenLogicalDotsPerInchChange.

Reply:
I implemented those two APIs originally and it did not seem to work correctly. It wasn't just that the correct scale factor wasn't used, it was that different things (icons, text, etc.) were scaled with different factors when I tried that. Even when I returned values that were integer multiples of each other, I still did not get the nice results that this provides. What should I do about that?

Reply:
Qt applications need to manually activate High-DPI scaling support, apparently the ones you tried don't do that. Qt 6 has High-DPI support always active. `QT_ENABLE_HIGHDPI_SCALING=1` should override that unless the application explicitly disables High-DPI scaling support (available only in Qt 5 API though).